packages/components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useMemo, useRef, useState } from "react";
+import { LayoutRectangle, TextInput, View } from "react-native";
+import { ScrollView } from "react-native-gesture-handler";
+import { useStyles } from "../../hooks/useStyles";
+import { EDSStyleSheet } from "../../styling";
+import { IconButton } from "../Button/IconButton";
+import { Input, InputProps } from "../Input";
+import { Menu } from "../Menu";
+
+type SingleSelect = {
+    multiple?: false;
+    /**
+     * A callback method invoked when the user selects an option from the autocomplete.
+     */
+    onSelect: (value: string) => void;
+};
+
+type MultiSelect = {
+    multiple: true;
+    /**
+     * A callback method invoked when the user selects an option from the autocomplete.
+     */
+    onSelect: (value: string[]) => void;
+};
+
+type AutocompleteProps = {
+    /**
+     * Options list that will be displayed in the autocomplete.
+     */
+    options: string[];
+    /**
+     * A label above the input field.
+     */
+    label?: string;
+    /**
+     * A placeholder for the input field.
+     */
+    placeholder?: string;
+} & InputProps &
+    (SingleSelect | MultiSelect);
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
```

Comment:
We don't need the React.FC wrapper around the types. This makes it so that the component accepts children, which I guess this does not. Look at the other types for inspiration.